lib/pure/httpclient.nim

```diff
@@ -1037,33 +1033,86 @@ proc request*(client: HttpClient | AsyncHttpClient, url: string,
   ##
   ## You need to make sure that the ``url`` doesn't contain any newline
   ## characters. Failing to do so will raise ``AssertionDefect``.
-  doAssert(not url.contains({'\c', '\L'}), "url shouldn't contain any newline characters")
+  ## **Deprecated since v1.5**: use HttpMethod enum instead; string parameter httpMethod is deprecated
+  when url is string:
+    doAssert(not url.contains({'\c', '\L'}), "url shouldn't contain any newline characters")
```

Comment:
This seems like it should be thrown by `parseUri` below rather than handled here.

Reply:
agreed, should be handled by uri.

Reply:
should be done in a seperate PR on uri module, for now I leave it as is.

Reply:
Instead of guessing intentions let's do what we deem right. Otherwise we will end up with code that nobody understands except the original person who is no longer around to answer for it :)

Reply:
Since this is the only thing I think needs to be changed though, can you create a separate PR for this? I will merge this.